completed competency grader challenge -Xiaoxi

xiaoxixu opened the pull request:
could not find the exact button colors shown in demo

Commit:
completed competency grader challenge

Commit:
fixed coding style

Power-tile commented:
Nice work!
- One feature request: when no grade is selected, the "Grade" on top should be shown as ```0%``` instead of ```0```.
- One PR element: write a summary of this pull request, as we discussed in your last pull request
- Actual coding style, implementation, and simpler solution: let's discuss this part in the workshop. Writing them out is not very convenient to communicate 😂

Commit:
fixed a display feature (0%)

xiaoxixu commented:
Changes:
- completed competency grader challenge and stored it in challenges/competency-grader/Xiaoxi
- fixed the display issue: the "Grade" on top is shown as 0% instead of 0.

Changed Files:
- vue-playground/challenges/competency-grader/Xiaoxi/competency-grader.html
- vue-playground/challenges/competency-grader/Xiaoxi/script.js

Commit:
simplified background color change logic; improved coding style

xiaoxixu commented:
Changes:

-  simplified background color change of the buttons 
-  improved coding style, including indentation, variable declaration, separation between style(html) and dynamic changes(js)

Changed Files:

- vue-playground/challenges/competency-grader/Xiaoxi/competency-grader.html
- vue-playground/challenges/competency-grader/Xiaoxi/script.js
